Source/hydro/advection_util.cpp

```diff
@@ -708,3 +708,68 @@ Castro::do_enforce_minimum_density(const Box& bx,
     }
   });
 }
+
+
+void
+Castro::enforce_reflect_states(const Box& bx, const int idir,
+                               Array4<Real> const& qm,
+                               Array4<Real> const& qp) {
+
+    // special care for reflecting BCs
+    const int* lo_bc = phys_bc.lo();
+    const int* hi_bc = phys_bc.hi();
+
+    const auto domlo = geom.Domain().loVect3d();
+    const auto domhi = geom.Domain().hiVect3d();
+
+    bool lo_bc_test = lo_bc[idir] == Symmetry;
+    bool hi_bc_test = hi_bc[idir] == Symmetry;
+
+    // normal velocity
+    const int QUN = QU + idir;
+
+    // this is a loop over interfaces
+
+    if (lo_bc_test) {
+
+        amrex::ParallelFor(bx,
+        [=] AMREX_GPU_HOST_DEVICE (int i, int j, int k)
+        {
+
+            // reset the left state at domlo(0) if needed -- it is outside the domain
```

Comment:
Comment should be updated since this could be any of the dimensions now

Reply:
fixed